### ISSUE TITLE
Calendar: Keyboard accessibility

### ISSUE THREAD

**Issue by emiravete** [primefaces/primereact#1830]
**Current behavior**
Unable to change the time only with the keyboard

**Expected behavior**
Allow to change hours, minutes ans seconds with the keyboard

**Reproduce**
- goto https://primefaces.org/primereact/showcase/#/calendar
- click on Time / 24h field
- Try to change the time. You can set the focus on the arrow up and down but when you type enter or space nothing to happen. It works only with the mouse.

* **React version:**
Version used on https://primefaces.org/primereact

* **PrimeReact version:**
Version used on https://primefaces.org/primereact

* **Browser:**  Chrome | Firefox

**Comment by melloware** [comment]
### Describe the bug

If using Calendar with showOnFocus = false there is no way to open the Calendar overlay with just using keyboard.

Same issue happens if using showOnFocus=true and hiding Overlay using Esc and like to reopen Overlay.

### Reproducer

_No response_

### PrimeReact version

8.7.2

### React version

18.x

### Language

ALL

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome 107

### Steps to reproduce the behavior

1. Focus <Calendar showOnFocus={false} />
2. press Enter or ArrowDown

### Expected behavior

Calendar overlay is shown

**Comment by habubey** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog.
